server/parsec/components/auth.py

```diff
@@ -70,12 +73,24 @@ class AuthenticatedAuthInfo:
 
 
 class BaseAuthComponent:
-    def __init__(self, config: BackendConfig):
+    def __init__(self, event_bus: EventBus, config: BackendConfig):
         self._config = config
         self._device_cache: dict[
             tuple[OrganizationID, DeviceID],
             tuple[DateTime, AuthenticatedAuthInfo | AuthAuthenticatedAuthBadOutcome],
         ] = {}
+        event_bus.connect(self._on_event)
+
+    def _on_event(self, event: Event) -> None:
+        match event:
+            # Revocation and freezing/unfreezing affect the authentication process,
+            # so we clear the cache when such events occur.
+            case EventUserUnFrozen() | EventUserRevokedOrFrozen():
+                self._device_cache = {
+                    (org_id, device_id): v
+                    for ((org_id, device_id), v) in self._device_cache.items()
+                    if org_id != event.organization_id or device_id.user_id != event.user_id
+                }
```

Comment:
Why not simply remove the corresponding entry?
```suggestion
                self._device_cache.pop((event.organization_id, event.user_id), None)
```

Reply:
`self._device_cache` contains keys of `tuple[OrganizationID, DeviceID]`, but the event provide `OrganizationID` and `UserID`

We filter the devices in the cache on the `UserID` then

Reply:
Erh sorry I read too fast